frontend/src/Components/PrivateRouteGate.jsx

```diff
@@ -0,0 +1,30 @@
+/*
+ANIME BENSALEM, BRIAHANA MAUGÉ, JOSEPH P. PASAOA
+PrivateRouteGate Component | Capstone App (Pursuit Volunteer Mgr)
+*/
+
+
+/* IMPORTS */
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+
+
+const PrivateRouteGate = (props) => {
+  // this show mechanism is needed to hide private elements before redirect & after clicking logout
+  let showPrivateChildren = null;
+  if (props.isUserStateReady) {
+    showPrivateChildren = props.children;
+  }
+
+
+  return(
+    <>
+      {props.currentRole
+        ? showPrivateChildren
+        : <Redirect to={{ pathname: '/', state: { from: props.location }}} />
```

Comment:
Why did you use `to={{ pathname: '/', state: { from: props.location }}}` and not just `to='/'` ?

Reply:
for a future ux feature. with that state being passed, if someone were to bookmark an inside page/route but have to log back in, we could use that to push them to the page they had bookmarked and intended on going to after they log back in. i mightve included it already because i didnt want to go to trello and spend the minutes adding it to the future ideas card lol